fsspec/implementations/hdfs.py

```diff
@@ -95,94 +241,52 @@ def _open(
-            out["port"] = ops["port"]
-        return out
+        match = CHECKSUM_REGEX.match(result)
+        if match is None:
+            return None
 
-    def close(self):
-        self.pahdfs.close()
+        return match.group("checksum")
 
-    @classmethod
-    def _strip_protocol(cls, path):
-        ops = infer_storage_options(path)
-        return ops["path"]
+    def _run_command(self, cmd, env=None):
+        cmd = "hadoop fs -" + cmd
+        if self.user:
+            cmd = f"HADOOP_USER_NAME={self.user} " + cmd
 
-    def __getattribute__(self, item):
-        if item in [
-            "_open",
-            "close",
-            "__init__",
-            "__getattribute__",
-            "__reduce_ex__",
-            "open",
-            "ls",
-            "makedirs",
-        ]:
-            # all the methods defined in this class. Note `open` here, since
-            # it calls `_open`, but is actually in superclass
-            return lambda *args, **kw: getattr(PyArrowHDFS, item)(self, *args, **kw)
-        if item == "__class__":
-            return PyArrowHDFS
-        d = object.__getattribute__(self, "__dict__")
-        pahdfs = d.get("pahdfs", None)  # fs is not immediately defined
-        if pahdfs is not None and item in [
-            "chmod",
-            "chown",
-            "user",
-            "df",
-            "disk_usage",
-            "download",
-            "driver",
-            "exists",
-            "extra_conf",
-            "get_capacity",
-            "get_space_used",
-            "host",
-            "is_open",
-            "kerb_ticket",
-            "strip_protocol",
-            "mkdir",
-            "mv",
-            "port",
-            "get_capacity",
-            "get_space_used",
-            "df",
-            "chmod",
-            "chown",
-            "disk_usage",
-            "download",
-            "upload",
-            "_get_kwargs_from_urls",
-            "read_parquet",
-            "rm",
-            "stat",
-            "upload",
-        ]:
-            return getattr(pahdfs, item)
+        # NOTE: close_fds doesn't work with redirected stdin/stdout/stderr.
+        # See https://github.com/iterative/dvc/issues/1197.
+        close_fds = os.name != "nt"
+
+        executable = os.getenv("SHELL") if os.name != "nt" else None
+        p = subprocess.Popen(
+            cmd,
+            shell=True,
+            close_fds=close_fds,
+            executable=executable,
+            env=env or os.environ,
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        out, err = p.communicate()
+
+        if p.returncode != 0:
+            raise subprocess.CalledProcessError(p.returncode, cmd, out, err)
         else:
-            # attributes of the superclass, while target is being set up
-            return super().__getattribute__(item)
+            return out.decode("utf-8")
+
+    def _as_url(self, path):
```

Comment:
I'm not sure we need this anymore, but it may be worth considering having something like this in AbstractFileSystem (c.f., [`_unstrip_protocol`](https://github.com/intake/filesystem_spec/pull/744/files#diff-3f0d077f95c8b71995c7d1c0815d95d64aaa5c516411caa8a91b6f783867d86aR452) , which doesn't include any netloc things, and doesn't know about chained FSs).